File: app/src/main/java/com/davismiyashiro/expenses/view/opentab/ExpenseInterfaces.java
package com.davismiyashiro.expenses.view.opentab;

import com.davismiyashiro.expenses.datatypes.Expense;
import com.davismiyashiro.expenses.datatypes.Tab;

import java.util.List;

/**
 * @Author Davis Miyashiro
 */

public interface ExpenseInterfaces {
    interface ExpenseView {

        void showExpenses (List<Expense> expenses);
    }

    interface UserActionsListener {

        void setExpenseView (ExpenseView view);

        void loadExpenses (Tab tab);

        void removeExpense (Expense expense);

    }
}

File: app/src/main/java/com/davismiyashiro/expenses/view/opentab/ExpenseInterfaces.kt
package com.davismiyashiro.expenses.view.opentab

import com.davismiyashiro.expenses.datatypes.Expense
import com.davismiyashiro.expenses.datatypes.Tab

/**
 * @Author Davis Miyashiro
 */

interface ExpenseInterfaces {
    interface ExpenseView {

        fun showExpenses(expenses: MutableList<Expense>)
    }

    interface UserActionsListener {

        fun setExpenseView(view: ExpenseView)

        fun loadExpenses(tab: Tab)

        fun removeExpense(expense: Expense)

    }
}
